spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/actions/DeleteOrphanFilesSparkAction.java

```diff
@@ -234,6 +241,98 @@ private String jobDesc() {
     return String.format("Deleting orphan files (%s) from %s", optionsAsString, table.name());
   }
 
+  private boolean streamResults() {
+    return PropertyUtil.propertyAsBoolean(options(), STREAM_RESULTS, STREAM_RESULTS_DEFAULT);
+  }
+
+  private DeleteOrphanFiles.Result doExecute() {
+    Dataset<FileURI> actualFileIdentDS = actualFileIdentDS();
+    Dataset<FileURI> validFileIdentDS = validFileIdentDS();
+
+    SetAccumulator<Pair<String, String>> conflicts = new SetAccumulator<>();
+    spark().sparkContext().register(conflicts);
+
+    Dataset<String> orphanFileDS =
+        findOrphanFilesAsDataset(actualFileIdentDS, validFileIdentDS, conflicts);
+
+    if (streamResults()) {
+      return deleteFilesStreaming(orphanFileDS, conflicts);
+    } else {
+      List<String> orphanFiles = orphanFileDS.collectAsList();
+      validateNoConflicts(conflicts);
+      deleteFilesCollected(orphanFiles);
+      return ImmutableDeleteOrphanFiles.Result.builder().orphanFileLocations(orphanFiles).build();
+    }
+  }
+
+  /**
+   * Deletes orphan files in streaming mode using toLocalIterator(). Files are processed one
+   * partition at a time to avoid collecting all file paths to driver memory. Returns a sample of
+   * file paths (up to MAX_ORPHAN_FILE_PATHS_TO_RETURN_WHEN_STREAMING) plus a summary row as the
+   * last row.
+   *
+   * @param orphanFileDS dataset of file paths to delete
+   * @param conflicts accumulator for tracking prefix mismatches
+   * @return result with sample file paths and optional summary row
+   */
+  private DeleteOrphanFiles.Result deleteFilesStreaming(
+      Dataset<String> orphanFileDS, SetAccumulator<Pair<String, String>> conflicts) {
+    // Cache the dataset and force computation to populate the conflicts accumulator
+    // This allows us to validate conflicts before starting any deletions
+    orphanFileDS = orphanFileDS.cache();
```

Comment:
I'm not familiar with Spark, so this is just a question:
What is the cost of this? Do we lose what we gain with not reading every file at once?